lib/Doctrine/ODM/PHPCR/DocumentManager.php

```diff
@@ -925,4 +925,16 @@ public function getNodeForDocument($document)
 
         return $this->session->getNode($path);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDocumentId($document)
+    {
+        try {
+            return $this->unitOfWork->getDocumentId($document);
+        } catch (PHPCRException $e) {
```

Comment:
why eat away the exception? i think i would prefer to have exceptions here as well, and let the user handle them. otherwise people will be confused why the id is null when they ask for the id of the wrong object.